corehq/ex-submodules/pillowtop/run_pillowtop.py

```diff
@@ -17,24 +24,46 @@ def start_pillows(pillows=None):
     """
     run_pillows = pillows or get_all_pillow_instances()
 
-    try:
-        while True:
-            jobs = []
-            print("[pillowtop] Starting pillow processes")
-            for pillow_class in run_pillows:
-                p = multiprocessing.Process(target=pillow_class.run)
-                p.start()
-                jobs.append(p)
-            print("[pillowtop] all processes started, pids: %s" % ([x.pid for x in jobs]))
-            for j in jobs:
-                j.join()
-            print("[pillowtop] All processes complete, restarting")
-    except KeyboardInterrupt:
-        sys.exit()
+    while True:
+        pids = []
+        handler = functools.partial(_remove_prometheus_metric_files, pids)
+        with SignalHandlerContext([signal.SIGINT, signal.SIGTERM, signal.SIGHUP], handler):
+            try:
+                print("[pillowtop] Starting pillow processes")
+                jobs = []
+                for pillow_class in run_pillows:
+                    p = multiprocessing.Process(target=pillow_class.run)
+                    p.start()
+                    jobs.append(p)
+
+                pids.extend(x.pid for x in jobs)
+                print(f"[pillowtop] all processes started, pids: {pids}")
+
+                for j in jobs:
+                    j.join()
+            finally:
+                handler()
```

Comment:
Does the `with SignalHandlerContext` context manager make the try/finally redundant? Maybe not—I guess `SignalHandlerContext` doesn't call `handler()` on an `Exception`, just on those particular signals